src/Compiler/AbstractIL/il.fs

```diff
@@ -55,10 +55,10 @@ type PrimaryAssembly =
     static member IsPossiblePrimaryAssembly(fileName: string) =
         let name = System.IO.Path.GetFileNameWithoutExtension(fileName)
 
-        String.Compare(name, "mscorlib", true) <> 0
```

Comment:
I changed `<>` to `=` here. Or I did misunderstand something. This was introduced in https://github.com/dotnet/fsharp/pull/13870

Reply:
https://github.com/dotnet/fsharp/blob/b4afa997c703f2ed3e29ab4da1815b0090ecf514/src/Compiler/Driver/CompilerImports.fs#L2437-L2447
based on calling context this bug was resulting in making any framework assembly as primary assembly equivalent, this may affect some esoteric .NET implementations.